src/eduid_webapp/security/views/webauthn.py

```diff
@@ -0,0 +1,165 @@
+
+
+@webauthn_views.route('/register/complete', methods=['POST'])
+@UnmarshalWith(WebauthnRegisterRequestSchema)
+@MarshalWith(SecurityResponseSchema)
+@require_user
+def registration_complete(user, credential_id, attestation_object, client_data, description):
+    security_user = SecurityUser.from_user(user, current_app.private_userdb)
+    server = get_webauthn_server()
+
+    attestation = base64.b64decode(attestation_object)
+    att_obj = AttestationObject(attestation)
+    client_data = base64.b64decode(client_data)
+    cdata_obj = ClientData(client_data)
+    state = session['_webauthn_state_']
+    auth_data = server.register_complete(state, cdata_obj, att_obj)
+
+    cred_data = auth_data.credential_data
+    current_app.logger.debug('Proccessed Webauthn credential data: {}.'.format(cred_data))
+
+    credential = Webauthn(
+        keyhandle = credential_id,
+        credential_data = base64.b64encode(cred_data).decode('ascii'),
+        app_id = current_app.config['FIDO2_RP_ID'],
+        attest_obj = base64.b64encode(attestation).decode('ascii'),
+        description = description,
+        application = 'security'
+        )
+
+    security_user.credentials.add(credential)
+    save_and_sync_user(security_user)
+    current_app.stats.count(name='webauthn_register_complete')
+    current_app.logger.info('User {} has completed registration of a webauthn token'.format(security_user))
+    return {
+        'message': 'security.webauthn_register_success',
+        'credentials': compile_credential_list(security_user)
+    }
+
+
+@webauthn_views.route('/remove', methods=['POST'])
+@UnmarshalWith(RemoveWebauthnTokenRequestSchema)
+@MarshalWith(SecurityResponseSchema)
+@require_user
+def remove(user, credential_key):
+    security_user = SecurityUser.from_user(user, current_app.private_userdb)
+    token_to_remove = security_user.credentials.filter(Webauthn).find(credential_key)
+    if token_to_remove:
+        security_user.credentials.remove(credential_key)
+        save_and_sync_user(security_user)
+        current_app.stats.count(name='webauthn_token_remove')
+    return {
+        'message': 'security.webauthn-token-removed',
+        'credentials': compile_credential_list(security_user)
+    }
+
+
+@webauthn_views.route('/verify', methods=['POST'])
+@UnmarshalWith(VerifyWithWebauthnTokenRequestSchema)
+@MarshalWith(VerifyWithWebauthnTokenResponseSchema)
+@require_user
+def verify(user, key_handle, signature_data, client_data):
+    challenge = session.pop('_u2f_challenge_')
+    if not challenge:
+        current_app.logger.error('Found no U2F challenge data in session.')
+        return {'_error': True, 'message': 'security.u2f.missing_challenge_data'}
+    data = {
+        'keyHandle': key_handle,
+        'signatureData': signature_data,
+        'clientData': client_data
+    }
+    device, c, t = complete_authentication(challenge, data, current_app.config['U2F_FACETS'])
+    current_app.stats.count(name='u2f_verify')
```

Comment:
u2f_verify -> webauthn_verify

Reply:
yep, changed

Reply:
I guess we don't really need the u2f views anymore after we have put webauthn in production so I would let this be for now.